code/modules/projectiles/guns/energy/rifle.dm

```diff
@@ -86,3 +86,14 @@
 		list(mode_name="lethal", projectile_type=/obj/item/projectile/beam, fire_sound='sound/weapons/Laser.ogg'),
 		list(mode_name="DESTROY", projectile_type=/obj/item/projectile/beam/pulse, fire_sound='sound/weapons/pulse.ogg', fire_delay=15, charge_cost=400)
 		)
+
+/obj/item/weapon/gun/energy/rifle/laser/xray
```

Comment:
Wallbanging. It needs to wallbang. Alberyk, please tell me that it can wallbang.